File: Data Structures & Algorithms/top-k-elements-in-list/submission-1.java
class Solution {
    public int[] topKFrequent(int[] nums, int k) {
        HashMap<Integer, Integer> hash = new HashMap<>();
        for (int i : nums) {
            hash.put(i, hash.getOrDefault(i, 0) + 1);
        }
        System.out.println(hash);
        PriorityQueue<Integer> max_heap = new PriorityQueue<>((a, b) -> hash.get(b) - hash.get(a));
        max_heap.addAll(hash.keySet());
        int[] result = new int[k];
        for (int i = 0 ; i < k ; i++) {
            result[i] = max_heap.poll();
        }
        return result;
    }
}

File: Data Structures & Algorithms/duplicate-integer/submission-1.go
func hasDuplicate(nums []int) bool {
    hash_map := make(map[int]int)
    for _, v := range nums {
        _, ok := hash_map[v]
        if ok {
            return true;
        } else {
            hash_map[v] = 1
        }
    }
    return false;
}
